pkg/capacity/types/pool.go

```diff
@@ -255,6 +255,24 @@ func UpdatePool(ctx context.Context, db *mongo.Database, pool Pool) error {
 	return nil
 }
 
+// GetPools gets all pools from the database
+func GetPools(ctx context.Context, db *mongo.Database) ([]Pool, error) {
+	col := db.Collection(CapacityPoolCollection)
+
+	cur, err := col.Find(ctx, options.FindOptions{})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to load pool list")
+	}
+
+	defer cur.Close(ctx)
+	out := []Pool{}
+	if err := cur.All(ctx, &out); err != nil {
```

Comment:
This call is gonna be a killer if we have millions of pools. You should instead implement paging or streams. I would instead return a channel and go over the cursor item by item to end or until context is done.